agent/grpc/handler.go

```diff
@@ -11,15 +11,16 @@ import (
 )
 
 // NewHandler returns a gRPC server that accepts connections from Handle(conn).
-func NewHandler(addr net.Addr) *Handler {
+// The register function will be called with the grpc.Server to register
+// gRPC services with the server.
+func NewHandler(addr net.Addr, register func(server *grpc.Server)) *Handler {
 	// We don't need to pass tls.Config to the server since it's multiplexed
 	// behind the RPC listener, which already has TLS configured.
 	srv := grpc.NewServer(
 		grpc.StatsHandler(newStatsHandler()),
 		grpc.StreamInterceptor((&activeStreamCounter{}).Intercept),
 	)
-
-	// TODO(streaming): add gRPC services to srv here
+	register(srv)
```

Comment:
It might be due to a lack of context on my part but I don't understand why this function should execute the register func instead of allowing the caller a way to register after creation.

Wouldn't it be cleaner to just export the `Srv` field in the `Handler` and allow the caller of this function to register endpoints with the server after the fact.

Reply:
Exposing the field could also work, and I think ends up being not all that much different, since the caller could technically capture the srv using the register function (as one test in this PR does).

I like the register func approach because I think it makes the usage more obvious. From the function definition you can see that it requires a register function to add services, and the return type has 3 exported methods that can be used.

If the returned type expose `Srv` it's less clear to the caller what needs to be done. They could call `Handler.Srv.Serve()` directly with the wrong listener, which would cause `Handle()` to block and never unblock. 

Ideally I would have had the `register` function take an interface with only the `RegisterService` call on it, but unfortunately the gRPC `RegisterXServer` functions accept the concrete type and not an interface, so that isn't an option. (It could actually work if we use the `_X_serviceDesc` vars that gRPC exposes, but the naming of those makes me believe they are probably intended to be implementation details and may change when we upgrade).

Reply:
I suspected it might be a dependent library as the reason for not being able just use a func on the Handler itself. In that case this is fine.